llvm/lib/Target/RISCV/MCTargetDesc/RISCVInstPrinter.cpp

```diff
@@ -256,6 +278,8 @@ void RISCVInstPrinter::printSpimm(const MCInst *MI, unsigned OpNo,
   if (Opcode == RISCV::CM_PUSH)
     Spimm = -Spimm;
 
+  // RAII guard for ANSI color escape sequences
```

Comment:
I don't understand. Why is RAII relevant here? Surely markup isn't side-effecting and is just a filter on the ostream?

Reply:
`markup` simply returns a temporary wrapper `WithMarkup`. It set/reset the foreground color of the terminal in its ctor/dtor.

Reply:
Someone else pointed out that this was a little confusing. For the ARM backend (#65561) I did the following to make it more obvious: 

```
WithMarkup ScopedMarkup = markup(O, Markup::Immediate);
...
```